### ISSUE TITLE
unused_must_use: comparison inside macro will cause invalid code to be suggested

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust#104913]
Given the following code: 

```rust
macro_rules! negative {
      ($e:expr) => { $e < 0 }
}

fn main() {
      negative!(1 as i32);
}
```

The current output is:

```
warning: unused comparison that must be used
 --> src/main.rs:2:22
  |
2 |       ($e:expr) => { $e < 0 }
  |                      ^^^^^^ the comparison produces a value
...
6 |       negative!(1 as i32);
  |       ------------------- in this macro invocation
  |
  = note: `#[warn(unused_must_use)]` on by default
  = note: this warning originates in the macro `negative` (in Nightly builds, run with -Z macro-backtrace for more info)
help: use `let _ = ...` to ignore the resulting value
  |
2 |       ($e:expr) => { let _ = $e < 0 }
  |                      +++++++
```
However, this does make it worse:

````
The following errors were reported:
error: macro expansion ends with an incomplete expression: expected one of `.`, `;`, `?`, `else`, or an operator
 --> src/main.rs:2:36
  |
2 |       ($e:expr) => { let _ = $e < 0 }
  |                                    ^ expected one of `.`, `;`, `?`, `else`, or an operator

error: aborting due to previous error
````

**Comment by jruderman** [comment]
Should it suggest the change in main instead? i.e.

`let _ = negative!(1 as i32);`
